File: integrations/php/oicana-php/src/CompilationMode.php
<?php

declare(strict_types=1);

namespace Oicana;

/**
 * Compilation mode for template rendering.
 *
 * - Development: Uses default values for inputs when not provided
 * - Production: Requires all inputs to be explicitly provided
 */
enum CompilationMode: int
{
    case Production = 0;
    case Development = 1;

    /**
     * Convert to native extension value.
     */
    public function toNative(): int
    {
        return $this->value;
    }
}

File: integrations/java/oicana/src/main/java/com/oicana/CompilationMode.java
package com.oicana;

/**
 * The compilation mode for a template.
 */
public enum CompilationMode {
    /**
     * Production mode. Use this for generating final documents.
     */
    PRODUCTION(0),
    /**
     * Development mode. Use this for previews and during template development.
     */
    DEVELOPMENT(1);

    final int value;

    CompilationMode(int value) {
        this.value = value;
    }
}
